src/ts/entitysystem/core/Map.ts

```diff
@@ -6,6 +6,37 @@ module entityframework.map {
     export interface Asset {
         type : string,
         key : string,
+
+        createDOMElement(src : string) : HTMLElement;
```

Comment:
createDomElement does not belong on this level of the interface.  The Asset interface is serialize and describable.  When the asset is deserialized it does not have the createDOMElement method.